nexus/transport.go

```diff
@@ -0,0 +1,675 @@
+package nexus
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"maps"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+)
+
+// Transport is the low-level abstraction used by [Client] and [OperationHandle] to make network calls.
+//
+// NOTE: Experimental
+type Transport interface {
+
+	// StartOperation calls the configured Nexus endpoint to start an operation.
+	//
+	// This method has the following possible outcomes:
+	//
+	//  1. The operation completes successfully. The result of this call will be set as a [LazyValue] in
+	//     TransportStartOperationResponse.Complete and must be consumed to free up the underlying connection.
+	//
+	//	2. The operation completes unsuccessfully. The response will contain an [OperationError] in
+	//	   TransportStartOperationResponse.Complete
+	//
+	//  3. The operation was started and the handler has indicated that it will complete asynchronously. An
+	//     [OperationHandle] will be returned as TransportStartOperationResponse.Pending, which can be used to perform actions
+	//     such as getting its result.
+	//
+	//  4. There was an error making the call. The returned response will be nil and the error will be non-nil.
+	//     Most often it will be a [HandlerError].
+	//
+	// NOTE: Experimental
+	StartOperation(ctx context.Context, input any, options TransportStartOperationOptions) (*TransportStartOperationResponse[*LazyValue], error)
+
+	// GetOperationInfo returns information about a specific operation. Should only be called through an
+	// [OperationHandle].
```

Comment:
I would consider removing these comment on what should only be called through an `OperationHandle`. This entire interface is considered low level AFAIC and should typically not be called directly.